src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java

```diff
@@ -131,4 +173,167 @@ public void joinCommonsTest() throws Exception {
+        .userId(1L)
+        .commonsId(2L)
+        .totalWealth(0)
+        .build();
+
+    String requestBody = mapper.writeValueAsString(uc);
+
+    //Instead of returning empty, we instead say that it already exists. We shouldn't create a new entry.
+    when(userCommonsRepository.findByCommonsIdAndUserId(2L,1L)).thenReturn(Optional.of(uc));
+    when(userCommonsRepository.save(eq(uc))).thenReturn(uc);
+    
+    when(commonsRepository.findById(eq(2L))).thenReturn(Optional.of(c));
+
+    MvcResult response = mockMvc
+        .perform(post("/api/commons/join?commonsId=2").with(csrf()).contentType(MediaType.APPLICATION_JSON)
+            .characterEncoding("utf-8").content(requestBody))
+        .andExpect(status().isOk()).andReturn();
+
+    verify(userCommonsRepository, times(1)).findByCommonsIdAndUserId(2L, 1L);
+
+    String responseString = response.getResponse().getContentAsString();
+    String cAsJson = mapper.writeValueAsString(c);
+
+    assertEquals(responseString, cAsJson);
+  }
+
+  @WithMockUser(roles = { "USER" })
+  @Test
+  public void user_commons_exists_but_commons_doesnt_test() throws Exception {
+    UserCommons uc = UserCommons.builder()
+        .userId(1L)
+        .commonsId(2L)
+        .totalWealth(0)
+        .build();
+
+    String requestBody = mapper.writeValueAsString(uc);
+
+    //UserCommons exists, but somehow the commons it was linked to has been deleted.
+    when(userCommonsRepository.findByCommonsIdAndUserId(2L,1L)).thenReturn(Optional.of(uc));
+    when(userCommonsRepository.save(eq(uc))).thenReturn(uc);
+    
+    when(commonsRepository.findById(eq(2L))).thenReturn(Optional.empty());
+
+    MvcResult response = mockMvc
+        .perform(post("/api/commons/join?commonsId=2").with(csrf()).contentType(MediaType.APPLICATION_JSON)
+            .characterEncoding("utf-8").content(requestBody))
+        .andExpect(status().is(404)).andReturn();
+
+    verify(userCommonsRepository, times(1)).findByCommonsIdAndUserId(2L, 1L);
+
+    String responseString = response.getResponse().getContentAsString();
+
+    if(responseString.charAt(2) == 'm'){
+      assertEquals("{\"message\":\"Commons with id 2 not found\",\"type\":\"EntityNotFoundException\"}", responseString); 
+    }else{
+      assertEquals("{\"type\":\"EntityNotFoundException\",\"message\":\"Commons with id 2 not found\"}", responseString); 
+    }
+  }
+
+  @WithMockUser(roles = { "USER" })
+  @Test
+  public void join_and_create_userCommons_for_nonexistent_commons() throws Exception {
+    UserCommons uc = UserCommons.builder()
+        .userId(1L)
+        .commonsId(2L)
+        .totalWealth(0)
+        .build();
+
+    UserCommons ucSaved = UserCommons.builder()
+        .id(17L)
+        .userId(1L)
+        .commonsId(2L)
+        .totalWealth(0)
+        .build();
+
+    String requestBody = mapper.writeValueAsString(uc);
+
+    when(userCommonsRepository.findByCommonsIdAndUserId(anyLong(),anyLong())).thenReturn(Optional.empty());
```

Comment:
These `anyLong()` arguments should be replaced with `2L` and `1L`.

Reply:
Nice catch. Will fix.